dynamic_rest/serializers.py

```diff
@@ -301,6 +317,29 @@ def get_name(cls):
         return cls.Meta.name
 
     @classmethod
+    def get_url(self, pk=None):
```

Comment:
@ryochiji this is a pivotal part of the PR as it allows relation-fields and serializers to easily reference their endpoint, even for non-canonical serializers.

But there are a couple of issues:
- This assumes that a serializer is going to be defined by a single viewset, which may not be the case. If a serializer is registered with multiple viewsets, the current behavior is to take the URL of the latest viewset (arbitrary)
- Potentially taxing on performance, although this method would only be used for self-links (disabled by default) and the admin view, so don't think this would cause regressions for the JSON API. Can also probably fold this into the serializer cache